src/utils/components/DiskModal/DiskFormFields/StorageClassAndPreallocation.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { FC, useMemo, useReducer } from 'react';
+
+import { diskReducerActions } from '../state/actions';
+import { initialStateDiskForm } from '../state/initialState';
+import { diskReducer } from '../state/reducers';
+import { requiresDataVolume } from '../utils/helpers';
+
+import useStorageProfileClaimPropertySets from './hooks/useStorageProfileClaimPropertySets';
+import AlertedStorageClassSelect from './StorageClass/AlertedStorageClassSelect';
+import { sourceTypes } from './utils/constants';
+import AccessMode from './AccessMode';
+import ApplyStorageProfileSettingsCheckbox from './ApplyStorageProfileSettingsCheckbox';
+import EnablePreallocationCheckbox from './EnablePreallocationCheckbox';
+import VolumeMode from './VolumeMode';
+
+const StorageClassAndPreallocation: FC = () => {
+  const [diskState, dispatchDiskState] = useReducer(diskReducer, initialStateDiskForm);
+
+  const sourceRequiresDataVolume = useMemo(
+    () => requiresDataVolume(diskState.diskSource),
+    [diskState.diskSource],
+  );
+
+  const { claimPropertySets, loaded: storageProfileLoaded } = useStorageProfileClaimPropertySets(
+    diskState?.storageClass,
+  );
+
+  return (
+    <>
+      {(sourceRequiresDataVolume || diskState.diskSource === sourceTypes.UPLOAD) && (
+        <>
```

Comment:
I think it would be nice to add an if statement above the return statement if we don't render this component for example:
```
if (!sourceRequiresDataVolume && diskState.diskSource !== sourceTypes.UPLOAD) return null;

return ...
```

Reply:
I totally agree!